geode-core/src/main/java/org/apache/geode/internal/cache/execute/AbstractExecution.java

```diff
@@ -152,7 +159,11 @@ public static byte getReexecuteFunctionState(byte fnState) {
     throw new InternalGemFireException("Wrong fnState provided.");
   }
 
-  protected AbstractExecution() {}
+  protected AbstractExecution() {
```

Comment:
Execution instances get created for every function invocation. Do you want to read the system property that often? Could it instead by on: InternalFunctionExecutionServiceImpl which is a singleton?

Reply:
I didn’t see that but I’ll look closer. I wasn’t too concerned in the PR with optimizing for the property read as I was on moving this smell out of connections.